### DISCUSSION TITLE
Is CAS - CAMS+KFintech supported

### DISCUSSION THREAD

**Discussion by rajneesh19** [codereverser/casparser#51] [Q&A]
Cams now offer consolidated CAS for CAMS + KFintech, is that statement supported?  I tried to parse consolidated CAS but got an error.

Error parsing pdf file :: Error parsing CAS header

**Comment by codereverser** [comment]
Yes. It should be. 

Is your file generated via the following options? 
![image](https://user-images.githubusercontent.com/8553055/133386320-01d8e51d-d13b-4fe6-af57-5ae4ceea1605.png)

**Reply by rajneesh19** [reply]
File is generated from same link but my statement type is "Summary".

**Reply by codereverser** [reply]
Summary statements are also supported; but it is not tested well. 

Does the header look like the following? 
![image](https://user-images.githubusercontent.com/8553055/133388082-2df5d6cf-3ae9-49d1-91bb-336b7867c336.png)

**Comment by rajneesh19** [comment]
Yes, it's the same header.

**Reply by codereverser** [reply]
Sorry, I'm not able to reproduce this. 😞 

Just tested it on the latest summary statements from three different accounts and it seems to be working fine. 

- Try using different parsers and see if it solves the issue 

```
# Install extra dependencies
pip install -U casparser[mupdf]

# Parse file using default mupdf parser (This will use mupdf only if PyMuPDF is installed via above command)
casparser filename -p password

# Parse file using fallback parser (pdfminer)
casparser filename -p password --force-pdfminer
```

Also please try generating a detailed statement from CAMS and see if it works.

**Comment by rajneesh19** [comment]
I tried again after installing extra dependencies but I am getting the same
error message.

I am able to parse CAS statements from CAMS, no issues with this.

I tried to parse a detailed consolidated statement (Cams+Kfintech), I was
able to parse the statement but in summary output it did not show a Fund
which is not serviced by CAMS.  I have 8 funds in the statement but I got
only 7 funds in output.

If it helps, Fund is a UTI Index fund and it's listed last in the statement.

**Reply by codereverser** [reply]
Oh. that's weird. Is the closing units 0.0 for the UTI index fund?

**Comment by rajneesh19** [comment]
No, it has funds and closing unit balance is not 0.

**Reply by codereverser** [reply]
This is weird. I'm not able to reproduce this bug. From the first look, CAMS statement format doesn't seem to have changed recently; so the parser should work. Most likely this is an issue with individual portfolio and needs to be manually investigated. 

Also can you please try the CAS statement from Kfintech and see if the issue is there too
https://mfs.kfintech.com/investor/General/ConsolidatedAccountStatement

**Reply by codereverser** [reply]
just following up on this; I have gone through a large amount of independent PDF files over the last couple of weeks and haven't been able to re-produce this issue. 

Would you mind sharing the broken CAS file with me (**privately via email**) for debugging? It is totally fine if not; shall wait for more similar bug reports and try to resolve it later.

**Comment by rajneesh19** [comment]
Sorry, CAS has a lot of personal and financial data so I am not comfortable
sharing that.

**Reply by codereverser** [reply]
Yeah. I totally understand. 

kfintech and CAMS have come up with a new website recently @ https://mfcentral.com which has most of the features of casparser. it displays portfolio overview without needing to import CAS PDF files. You can give it a try.

**Comment by rajneesh19** [comment]
Yes , I have signed up on mfcentral.com and it looks promising.  I use Google
sheet
<https://freefincal.com/google-spreadsheet-portfolio-tracker-stocks-mutual-funds/>to
track my Portfolio and wanted to have some automated/semi automated way to
get units of MF where I run SIPs updated on Sheet on periodic basis.
That's why I started looking at CAS parsers.
